runtime/oti/j9nonbuilder.h

```diff
@@ -2084,6 +2084,8 @@ typedef struct J9ConstantPool {
 #define J9CPTYPE_METHOD_TYPE  13
 #define J9CPTYPE_METHODHANDLE  14
 #define J9CPTYPE_ANNOTATION_UTF8  15
+#define J9CPTYPE_INTERFACE_STATIC_METHOD 16
```

Comment:
These need a comment to describe when they will be used and why they need to be different than the existing static / instance method cp types.

Both of these tags will always result in an error at resolve - we should call that out as well.

Reply:
I am not sure what you mean by both tags will always result in error, they are checked at resolve time, which gives `IncompatibleClassChangeError` if the check failed. Otherwise the resolve should succeed. did I miss some check?

Reply:
Sorry, you're right.  This is for the case were there is an `invokestatic` targeting an `InterfaceMethodRef` which requires that the resolved class is an interface.  Do I have that right now?

The comment added here should explain **why** these new types are needed.